### PR TITLE
GEOMESA-947 Add SI Suffixes for GeoMesa Plugin

### PR DESCRIPTION
**Pull request by jahhulbert-ccri**
Signed-off-by: Andrew Hulbert <andrew.hulbert@ccri.com>

### CODE REVIEW

**Comment by jnh5y** [comment]
Looks good to me.  Wait for CI and then merge?

File: geomesa-plugin/src/main/resources/org/locationtech/geomesa/plugin/ui/js/entries-chart.js [hunk 1]
```diff
@@ -17,10 +17,20 @@ function entriesChart() {
         .append("g")
         .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
 
+    var suffixTickFormatter = function(n) {
+        var suffixes = ["", "K", "M", "G", "T", "P", "E"];
```
Comment (cne1x) [inline geomesa-plugin/src/main/resources/org/locationtech/geomesa/plugin/ui/js/entries-chart.js]: No "Z"?  C'mon, man:  We're not in the kiddie pool any longer.

File: geomesa-plugin/src/main/resources/org/locationtech/geomesa/plugin/ui/js/entries-chart.js [hunk 2]
```diff
@@ -17,10 +17,20 @@ function entriesChart() {
...
+        var idx = 0;
+        while (n >= 1000.0) {
+            idx += 1;
+            n = (1.0 * n) / 1000.0;
```
Comment (cne1x) [inline geomesa-plugin/src/main/resources/org/locationtech/geomesa/plugin/ui/js/entries-chart.js]: [style] It feels like either this line or line 23 ought to change...  If ```n``` is ```Double```, then the multiplication by 1.0 shouldn't be necessary; if it's an ```Int```, then the test should be ```n >= 1000```, yes?

Comment (cne1x) [inline geomesa-plugin/src/main/resources/org/locationtech/geomesa/plugin/ui/js/entries-chart.js]: Never mind.  JavaScript.  Sigh.

File: geomesa-plugin/src/main/scala/org/locationtech/geomesa/plugin/ui/components/FeatureInfoPanel.scala [hunk 3]
```diff
@@ -35,19 +35,41 @@ class FeatureInfoPanel(id: String, featureData: FeatureData) extends Panel(id) {
...
+      item.add(new Label("metadataNumEntries", Model.of(Formatting.formatLargeNum(metadata.numEntries))))
+      item.add(new Label("metadataFileSize", Model.of(Formatting.formatMem(metadata.fileSize))))
 
     }
   })
 }
+
+object Formatting {
+  private val siSuffix = Array("", "K", "M", "G", "T", "P", "E")
+  private val memSuffix = siSuffix.take(1) ++ siSuffix.drop(1).map(_ + "B")
```
Comment (cne1x) [inline geomesa-plugin/src/main/scala/org/locationtech/geomesa/plugin/ui/components/FeatureInfoPanel.scala]: This leaves 0-999 without a suffix at all...  Wouldn't it be plain "B"?

File: geomesa-plugin/src/test/scala/org/locationtech/geomesa/plugin/ui/GeoMesaDataStoresPageTest.scala [hunk 4]
```diff
@@ -76,7 +76,7 @@ class GeoMesaDataStoresPageTest extends Specification {
       metadata.numEntries should be equalTo 450
       metadata.numSplits should be equalTo 100
       // exact file size varies slightly between runs... not sure why
-      Math.abs(metadata.fileSize - 0.026) should be lessThan 0.001
+      Math.abs(metadata.fileSize - 0.026*1024*1024) should be lessThan 0.001*1024*1024
```
Comment (cne1x) [inline geomesa-plugin/src/test/scala/org/locationtech/geomesa/plugin/ui/GeoMesaDataStoresPageTest.scala]: [style]  Does ```metadata.fileSize``` report size in mebibytes?  (Thanks for that Wikipedia reference, by the way.  Too much fun.)  I'm not certain what value this change adds.  Something like ```metadata.fileSize must be 0.026 +/- 0.001``` might be a bit more direct.

File: geomesa-plugin/src/main/resources/org/locationtech/geomesa/plugin/ui/js/entries-chart.js (hunk 1 above)
Comment (cne1x) [inline geomesa-plugin/src/main/resources/org/locationtech/geomesa/plugin/ui/js/entries-chart.js]: Okay, so a ```Long``` only goes up to 64 bits; still!

Comment (jahhulbert-ccri) [inline geomesa-plugin/src/main/resources/org/locationtech/geomesa/plugin/ui/js/entries-chart.js]: i know i had that in there when i was using Doubles then i switched to longs

File: geomesa-plugin/src/main/scala/org/locationtech/geomesa/plugin/ui/components/FeatureInfoPanel.scala (hunk 3 above)
Comment (jahhulbert-ccri) [inline geomesa-plugin/src/main/scala/org/locationtech/geomesa/plugin/ui/components/FeatureInfoPanel.scala]: hmmm that would be OK with me

File: geomesa-plugin/src/test/scala/org/locationtech/geomesa/plugin/ui/GeoMesaDataStoresPageTest.scala (hunk 4 above)
Comment (jahhulbert-ccri) [inline geomesa-plugin/src/test/scala/org/locationtech/geomesa/plugin/ui/GeoMesaDataStoresPageTest.scala]: looking...

File: geomesa-plugin/src/main/resources/org/locationtech/geomesa/plugin/ui/js/entries-chart.js (hunk 2 above)
Comment (jahhulbert-ccri) [inline geomesa-plugin/src/main/resources/org/locationtech/geomesa/plugin/ui/js/entries-chart.js]: TypeScript! TypeScript! TypeScript!

Comment (jnh5y) [inline geomesa-plugin/src/main/resources/org/locationtech/geomesa/plugin/ui/js/entries-chart.js]: Go ahead, fill out the CQs...

**Comment by cne1x** [comment]
Rebased on master, and merged as commit dacc81e4df5b7a772f886b375b2e28862d14ff6e

**Pull request closed without merging**